### PR TITLE
Flaky test: pause on IO error

### PR DESCRIPTION
**Pull request by brybacki**
**What this PR does / why we need it**:

The PR tries to find and fix the flakiness in one of the storage tests. 

- first commit - Make sure the actual vmi is used, and improve logging.
- second commit - fix the actual problem. 

The problem is that the test is using a virtual scsi device which randomly returns errors. The device sometimes fails during resize - before actually starting a VM. This is because kubevirt has `ExpandDisks` on, and the pvc has bigger size than the device. The fix creates the device and pvc with the same size so it does not need to be resized. 

```
{"component":"virt-launcher","level":"info","msg":"Domain name event: kubevirt-test-default3_testvmi-f9xqt","pos":"client.go:413","timestamp":"2022-05-19T10:13:31.756618Z"}
{"component":"virt-launcher","level":"error","msg":"internal error: unable to execute QEMU command 'block_resize': Cannot grow device files","pos":"qemuMonitorJSONCheckErrorFull:402","subcomponent":"libvirt","thread":"26","timestamp":"2022-05-19T10:13:31.757000Z"}
{"component":"virt-launcher","kind":"","level":"error","msg":"libvirt failed to expand disk image {disk  block {/dev/pvc-disk    pvc-disk \u003cnil\u003e} {virtio vdb }  0xc0007fa680 \u003cnil\u003e \u003cnil\u003e 0xc00082a5a0 \u003cnil\u003e \u003cnil\u003e \u003cnil\u003e virtio-non-transitional \u003cnil\u003e 0xc000316760 0xc0006ea700 true \u003cnil\u003e}","name":"testvmi-f9xqt","namespace":"kubevirt-test-default3","pos":"manager.go:957","reason":"virError(Code=1, Domain=10, Message='internal error: unable to execute QEMU command 'block_resize': Cannot grow device files')","timestamp":"2022-05-19T10:13:31.757366Z","uid":"063dddee-c169-4bc8-a39e-df9abc52bcc2"}
{"component":"virt-launcher","level":"info","msg":"kubevirt domain status: Paused(3):IOError(5)","pos":"client.go:288","timestamp":"2022-05-19T10:13:31.758798Z"}

```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #
This fails rarely, and when it fails it is always when trying to login to a VM for one test, and when checking the status for another. Maybe there is some kind of memory pressure or some problem with the parallel run. Below somple runs:

https://prow.ci.kubevirt.io/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7465/pull-kubevirt-e2e-k8s-1.22-sig-storage/1523712386618363904

https://prow.ci.kubevirt.io/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7465/pull-kubevirt-e2e-k8s-1.22-sig-storage/1523663268277653504

https://prow.ci.kubevirt.io/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7699/pull-kubevirt-e2e-k8s-1.22-sig-storage/1522643157060161536

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by xpivarc** [comment]
It might be helpful to include symptoms you are trying to treat and/or links.

**Comment by brybacki** [comment]
> It might be helpful to include symptoms you are trying to treat and/or links.

Thanks, edited the description to include such information.

**Comment by iholder-redhat** [comment]
Thanks for your changes!

> Maybe there is some kind of memory pressure or some problem with the parallel run.

The problem here AFAIK is that on creation the vmi object is fetched. Then, the vmi is being started and running. That causes the VMI object to change and update at etcd, but the login command uses and old and non-updated VMI. This is a recipe for problems.

/lgtm

**Comment by xpivarc** [comment]
FYI : 
[This](https://prow.ci.kubevirt.io/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7465/pull-kubevirt-e2e-k8s-1.22-sig-storage/1523712386618363904) shows that in test `Tests Suite: [sig-storage] Storage Starting a VirtualMachineInstance with error disk should pause VMI on IO error` the VMI is already paused([ref](https://storage.googleapis.com/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7465/pull-kubevirt-e2e-k8s-1.22-sig-storage/1523712386618363904/artifacts/k8s-reporter/2/1_vmis.log))
```
  {
                        "type": "Paused",
                        "status": "True",
                        "lastProbeTime": "2022-05-09T18:03:42Z",
                        "lastTransitionTime": "2022-05-09T18:03:42Z",
                        "reason": "PausedIOError",
                        "message": "VMI was paused, IO error"
                    }
```

**Comment by brybacki** [comment]
> FYI : [This](https://prow.ci.kubevirt.io/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7465/pull-kubevirt-e2e-k8s-1.22-sig-storage/1523712386618363904) shows that in test `Tests Suite: [sig-storage] Storage Starting a VirtualMachineInstance with error disk should pause VMI on IO error` the VMI is already paused([ref](https://storage.googleapis.com/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7465/pull-kubevirt-e2e-k8s-1.22-sig-storage/1523712386618363904/artifacts/k8s-reporter/2/1_vmis.log))
> 
> ```
>   {
>                         "type": "Paused",
>                         "status": "True",
>                         "lastProbeTime": "2022-05-09T18:03:42Z",
>                         "lastTransitionTime": "2022-05-09T18:03:42Z",
>                         "reason": "PausedIOError",
>                         "message": "VMI was paused, IO error"
>                     }
> ```

Yes, so it seems that assertion could not find the right VM. 
@xpivarc WDYT? Do you approve?

**Review by dhiller — APPROVED** [review]
/approve

Thanks!

**Comment by xpivarc** [comment]
> Yes, so it seems that assertion could not find the right VM.
@xpivarc WDYT? Do you approve?

It's failing to log in because the VMI is paused. I propose we enhance the `error` we return from `LogintTo*` functions...

**Comment by brybacki** [comment]
> > Yes, so it seems that assertion could not find the right VM.
> > @xpivarc WDYT? Do you approve?
> 
> It's failing to log in because the VMI is paused. I propose we enhance the `error` we return from `LogintTo*` functions...

Good idea

**Comment by brybacki** [comment]
It failed again. New reporting makes it easier to read. 

But it means there is some other error. As @xpivarc mentioned, the VM is already Paused on IO error.

Found this in the log:
```
{"component":"virt-launcher","level":"info","msg":"Domain name event: kubevirt-test-default3_testvmi-f9xqt","pos":"client.go:413","timestamp":"2022-05-19T10:13:31.756618Z"}
{"component":"virt-launcher","level":"error","msg":"internal error: unable to execute QEMU command 'block_resize': Cannot grow device files","pos":"qemuMonitorJSONCheckErrorFull:402","subcomponent":"libvirt","thread":"26","timestamp":"2022-05-19T10:13:31.757000Z"}
{"component":"virt-launcher","kind":"","level":"error","msg":"libvirt failed to expand disk image {disk  block {/dev/pvc-disk    pvc-disk \u003cnil\u003e} {virtio vdb }  0xc0007fa680 \u003cnil\u003e \u003cnil\u003e 0xc00082a5a0 \u003cnil\u003e \u003cnil\u003e \u003cnil\u003e virtio-non-transitional \u003cnil\u003e 0xc000316760 0xc0006ea700 true \u003cnil\u003e}","name":"testvmi-f9xqt","namespace":"kubevirt-test-default3","pos":"manager.go:957","reason":"virError(Code=1, Domain=10, Message='internal error: unable to execute QEMU command 'block_resize': Cannot grow device files')","timestamp":"2022-05-19T10:13:31.757366Z","uid":"063dddee-c169-4bc8-a39e-df9abc52bcc2"}
{"component":"virt-launcher","level":"info","msg":"kubevirt domain status: Paused(3):IOError(5)","pos":"client.go:288","timestamp":"2022-05-19T10:13:31.758798Z"}

```

/hold

**Comment by brybacki** [comment]
Ideas:
- Make test serial and turn off the `ExpandDisks` flag
- Create disk and pvc in such way that it does not need to be expanded.

**Comment by brybacki** [comment]
run 6 times without failures, will try again
/test pull-kubevirt-e2e-k8s-1.22-sig-storage
/test pull-kubevirt-e2e-k8s-1.23-sig-storage
/test pull-kubevirt-e2e-k8s-1.24-sig-storage

**Comment by brybacki** [comment]
and again it all passed

**Comment by brybacki** [comment]
@xpivarc I have implemented one of the ideas: "Create disk and pvc in such way that it does not need to be expanded.". I have uptaded the description, so it is ready for a final review.

**Comment by brybacki** [comment]
@xpivarc @mhenriks had to rebase on latest `main`

**Review by xpivarc — COMMENTED** [review]
/lgtm

File: tests/storage/storage.go [hunk 1]
```diff
@@ -138,17 +138,17 @@ var _ = SIGDescribe("Storage", func() {
 				tests.RemoveSCSIDisk(nodeName, address)
 			})
 
-			It("should pause VMI with error disk on IO error", func() {
+			It("[QUARANTINE] should pause VMI on IO error", func() {
```
Comment (xpivarc) [inline tests/storage/storage.go:141]: Do we want this?

**Review by xpivarc — COMMENTED** [review]
/hold

File: tests/storage/storage.go (hunk 1 above)
Comment (brybacki) [reply tests/storage/storage.go:141]: Well, it is still quarantined right? Until we prove it is stable.

Or do you mean you want to see a few runs without the label, and then add label again?

Comment (brybacki) [reply tests/storage/storage.go:141]: ok, It is a leftover from a rebase, let me remove this

Comment (brybacki) [reply tests/storage/storage.go]: removed

**Pull request merged**